dsgrid/registry/dimension_mapping_registry_manager.py

```diff
@@ -69,19 +69,27 @@ def dimension_manager(self, val: DimensionRegistryManager):
         self._dimension_mgr = val
 
     def _replace_duplicates(self, config: DimensionMappingsConfig):
+        def make_key(model):
+            return (
+                model.from_dimension.dimension_id,
+                model.to_dimension.dimension_id,
+                model.file_hash,
+            )
+
         hashes = {}
         for model in self.db.iter_models(all_versions=True):
-            hashes[model.file_hash] = model
+            key = make_key(model)
+            if key in hashes:
+                msg = f"Bug: the same file_hash exists in multiple mappings: {model.mapping_id} {key}"
+                raise Exception(msg)
```

Comment:
What is the user supposed to do if they hit this exception?

Reply:
And why is this a plain Exception? (When do we use plain Exceptions and when do we use dsgrid-named exceptions?)